### PR TITLE
Extract parallel planning from moveit cpp

### PR DESCRIPTION
**Pull request by sjahr**
### Description

This PR extracts the parallel planning from moveit_cpp. The motivation behind this is to make parallel planning more accessible for other components of the MoveIt ecosystem like MTC, the move_group interface, or hybrid planning.

Open TODOs
- [x] Update documentation
- [x] Update python bindings

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit2/blob/main/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://moveit.picknik.ai/humble/doc/examples/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit2/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_ros/planning/moveit_cpp/src/planning_component.cpp [hunk 1]
```diff
@@ -57,15 +57,6 @@ PlanningComponent::PlanningComponent(const std::string& group_name, const MoveIt
     RCLCPP_FATAL_STREAM(LOGGER, error);
     throw std::runtime_error(error);
   }
-  plan_request_parameters_.load(node_);
```
Comment (sjahr) [inline moveit_ros/planning/moveit_cpp/src/planning_component.cpp:60]: `plan_request_parameters_` was only used by `plan()` so I moved this code directly into the function and removed the member variable

File: moveit_ros/planning/planning_pipeline_interfaces/include/moveit/planning_pipeline_interfaces/plan_responses_container.hpp [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Sebastian Jahr
+   Desc: TODO */
```
Comment (henningkayser) [inline moveit_ros/planning/planning_pipeline_interfaces/include/moveit/planning_pipeline_interfaces/plan_responses_container.hpp:36]: TODO

File: moveit_ros/planning/planning_pipeline_interfaces/include/moveit/planning_pipeline_interfaces/planning_pipeline_interfaces.hpp [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Sebastian Jahr
+   Desc: TODO */
```
Comment (henningkayser) [inline moveit_ros/planning/planning_pipeline_interfaces/include/moveit/planning_pipeline_interfaces/planning_pipeline_interfaces.hpp:36]: TODO

File: moveit_ros/planning/planning_pipeline_interfaces/include/moveit/planning_pipeline_interfaces/solution_selection_functions.hpp [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Sebastian Jahr
+   Desc: TODO */
```
Comment (henningkayser) [inline moveit_ros/planning/planning_pipeline_interfaces/include/moveit/planning_pipeline_interfaces/solution_selection_functions.hpp:36]: TODO

File: moveit_ros/planning/planning_pipeline_interfaces/include/moveit/planning_pipeline_interfaces/stopping_criterion_functions.hpp [hunk 5]
```diff
@@ -33,16 +33,19 @@
  *********************************************************************/
 
 /* Author: Sebastian Jahr
-   Desc: Common callback functions for parallel planning */
+   Desc: TODO */
```
Comment (henningkayser) [inline moveit_ros/planning/planning_pipeline_interfaces/include/moveit/planning_pipeline_interfaces/stopping_criterion_functions.hpp:36]: TODO

File: moveit_ros/planning/planning_pipeline_interfaces/src/plan_responses_container.cpp [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Sebastian Jahr
+   Desc: TODO */
```
Comment (henningkayser) [inline moveit_ros/planning/planning_pipeline_interfaces/src/plan_responses_container.cpp:37]: TODO

**Comment by sjahr** [comment]
@peterdavidfagan Would you mind taking a look at this and confirming that it doesn't break the python bindings for parallel planning? Thanks in advance!

**Review by henningkayser — APPROVED** [review]

**Comment by peterdavidfagan** [comment]
Hi @sjahr, 

Apologies for the delay in response on this.

> @peterdavidfagan Would you mind taking a look at this and confirming that it doesn't break the python bindings for parallel planning? Thanks in advance!

Thanks for making the required changes yourself, on initial inspection it looks good to me. I will run the update against the latest set of tutorials and perform one final pass before confirming.

**Review by peterdavidfagan — APPROVED** [review]

**Comment by peterdavidfagan** [comment]
🚢

**Pull request merged**